x-pack/plugins/uptime/server/lib/alerts/status_check.ts

```diff
@@ -287,26 +315,67 @@ export const statusCheckAlertFactory: UptimeAlertTypeFactory<ActionGroupIds> = (
       };
 
       let downMonitorsByLocation: GetMonitorStatusResult[] = [];
+      let upMonitorsByLocation: GetMonitorStatusResult[] = [];
 
       // if oldVersionTimeRange present means it's 7.7 format and
       // after that shouldCheckStatus should be explicitly false
       if (!(!oldVersionTimeRange && shouldCheckStatus === false)) {
-        downMonitorsByLocation = await libs.requests.getMonitorStatus({
+        const queryConfig = {
           uptimeEsClient,
           timerange,
           numTimes,
           locations: [],
           filters: filterString,
-        });
+        };
+        downMonitorsByLocation = (await libs.requests.getMonitorStatus(queryConfig)) || [];
+        upMonitorsByLocation =
+          (await libs.requests.getMonitorStatus({
+            ...queryConfig,
+            status: 'up',
+          })) || [];
       }
+      /* Active down monitors which do not have a corresponding up document.
+       * Monitors should only be considered resolved when an up document is found */
+      const unresolvedDownMonitors = getUnresolvedDownMonitors(
+        upMonitorsByLocation,
+        state.activeDownMonitors
```

Comment:
we should pass this activeDownMonitors to get getMonitorStatus function as es filters.

Reply:
that will make that query much more performant.

Reply:
Is the idea to make use of this filter just for the up monitors query?

Reply:
yes, and when activeDownMonitors list is empty it will return any empty list, hence saving us an up monitors es query